flink-end-to-end-tests/pom.xml

```diff
@@ -255,6 +298,21 @@ under the License.
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
```

Comment:
do we still need this despite excluding all dependencies?

Reply:
You are right. Removing it.

Reply:
There are convergence issues on CI for Hadoop 2.8.3 (on 2.4.1 it converges)
But since we are not providing the flink-end-to-end-tests officially as a module for users, I decided to ignore convergence:

```
2020-05-11T10:54:23.9920427Z [INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (dependency-convergence) @ flink-end-to-end-tests ---
2020-05-11T10:54:24.0051160Z [WARNING] 
2020-05-11T10:54:24.0051590Z Dependency convergence error for com.google.guava:guava:11.0.2 paths to dependency are:
2020-05-11T10:54:24.0052711Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0053292Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0053794Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0054246Z       +-com.google.guava:guava:11.0.2
2020-05-11T10:54:24.0054558Z and
2020-05-11T10:54:24.0054967Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0055491Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0055971Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0056429Z       +-org.apache.hadoop:hadoop-auth:2.8.3
2020-05-11T10:54:24.0057162Z         +-org.apache.curator:curator-framework:2.7.1
2020-05-11T10:54:24.0057681Z           +-com.google.guava:guava:16.0.1
2020-05-11T10:54:24.0057931Z and
2020-05-11T10:54:24.0058355Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0058898Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0059397Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0059886Z       +-org.apache.curator:curator-client:2.7.1
2020-05-11T10:54:24.0060526Z         +-com.google.guava:guava:16.0.1
2020-05-11T10:54:24.0060915Z and
2020-05-11T10:54:24.0061396Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0061913Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0062417Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0062922Z       +-org.apache.curator:curator-recipes:2.7.1
2020-05-11T10:54:24.0063396Z         +-com.google.guava:guava:16.0.1
2020-05-11T10:54:24.0063643Z and
2020-05-11T10:54:24.0064067Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0064623Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0065096Z     +-org.apache.hadoop:hadoop-hdfs:2.8.3
2020-05-11T10:54:24.0065569Z       +-com.google.guava:guava:11.0.2
2020-05-11T10:54:24.0065811Z and
2020-05-11T10:54:24.0066236Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0066774Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0067274Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0067789Z       +-org.apache.hadoop:hadoop-yarn-api:2.8.3
2020-05-11T10:54:24.0068256Z         +-com.google.guava:guava:11.0.2
2020-05-11T10:54:24.0068501Z and
2020-05-11T10:54:24.0068940Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0069632Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0070133Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0070582Z       +-com.google.guava:guava:11.0.2
2020-05-11T10:54:24.0071129Z and
2020-05-11T10:54:24.0071673Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0072195Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0072692Z     +-org.apache.hadoop:hadoop-yarn-client:2.8.3
2020-05-11T10:54:24.0073146Z       +-com.google.guava:guava:11.0.2
2020-05-11T10:54:24.0073347Z 
2020-05-11T10:54:24.0073499Z [WARNING] 
2020-05-11T10:54:24.0074079Z Dependency convergence error for commons-beanutils:commons-beanutils:1.8.0 paths to dependency are:
2020-05-11T10:54:24.0074699Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0075216Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0075681Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0076191Z       +-commons-configuration:commons-configuration:1.7
2020-05-11T10:54:24.0076707Z         +-commons-digester:commons-digester:1.8.1
2020-05-11T10:54:24.0077208Z           +-commons-beanutils:commons-beanutils:1.8.0
2020-05-11T10:54:24.0077482Z and
2020-05-11T10:54:24.0078082Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0078619Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0079262Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0079773Z       +-commons-configuration:commons-configuration:1.7
2020-05-11T10:54:24.0080296Z         +-commons-beanutils:commons-beanutils:1.8.3
2020-05-11T10:54:24.0135511Z 
2020-05-11T10:54:24.0135752Z [WARNING] 
2020-05-11T10:54:24.0136471Z Dependency convergence error for org.codehaus.jackson:jackson-mapper-asl:1.8.3 paths to dependency are:
2020-05-11T10:54:24.0137294Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0137797Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0138241Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0138863Z       +-com.sun.jersey:jersey-json:1.9
2020-05-11T10:54:24.0139663Z         +-org.codehaus.jackson:jackson-mapper-asl:1.8.3
2020-05-11T10:54:24.0140120Z and
2020-05-11T10:54:24.0140860Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0141350Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0141803Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0142263Z       +-org.codehaus.jackson:jackson-mapper-asl:1.9.13
2020-05-11T10:54:24.0142514Z and
2020-05-11T10:54:24.0142897Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0143382Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0143835Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0144244Z       +-org.apache.avro:avro:1.8.2
2020-05-11T10:54:24.0144709Z         +-org.codehaus.jackson:jackson-mapper-asl:1.9.13
2020-05-11T10:54:24.0144946Z and
2020-05-11T10:54:24.0145341Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0145806Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0146253Z     +-org.apache.hadoop:hadoop-hdfs:2.8.3
2020-05-11T10:54:24.0146738Z       +-org.codehaus.jackson:jackson-mapper-asl:1.9.13
2020-05-11T10:54:24.0146971Z and
2020-05-11T10:54:24.0147368Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0147836Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0148302Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0148766Z       +-org.codehaus.jackson:jackson-mapper-asl:1.9.13
2020-05-11T10:54:24.0148987Z 
2020-05-11T10:54:24.0149127Z [WARNING] 
2020-05-11T10:54:24.0149652Z Dependency convergence error for org.codehaus.jackson:jackson-xc:1.8.3 paths to dependency are:
2020-05-11T10:54:24.0150218Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0151586Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0152106Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0152567Z       +-com.sun.jersey:jersey-json:1.9
2020-05-11T10:54:24.0153061Z         +-org.codehaus.jackson:jackson-xc:1.8.3
2020-05-11T10:54:24.0153587Z and
2020-05-11T10:54:24.0154075Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0154708Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0155178Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0155638Z       +-org.codehaus.jackson:jackson-xc:1.9.13
2020-05-11T10:54:24.0155824Z 
2020-05-11T10:54:24.0155963Z [WARNING] 
2020-05-11T10:54:24.0156494Z Dependency convergence error for org.codehaus.jackson:jackson-jaxrs:1.8.3 paths to dependency are:
2020-05-11T10:54:24.0157084Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0157549Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0158003Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0158413Z       +-com.sun.jersey:jersey-json:1.9
2020-05-11T10:54:24.0158869Z         +-org.codehaus.jackson:jackson-jaxrs:1.8.3
2020-05-11T10:54:24.0159109Z and
2020-05-11T10:54:24.0159494Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0159991Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0160444Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0160915Z       +-org.codehaus.jackson:jackson-jaxrs:1.9.13
2020-05-11T10:54:24.0161106Z 
2020-05-11T10:54:24.0161259Z [WARNING] 
2020-05-11T10:54:24.0161975Z Dependency convergence error for org.codehaus.jackson:jackson-core-asl:1.8.3 paths to dependency are:
2020-05-11T10:54:24.0162591Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0163069Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0163535Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0163977Z       +-com.sun.jersey:jersey-json:1.9
2020-05-11T10:54:24.0164608Z         +-org.codehaus.jackson:jackson-core-asl:1.8.3
2020-05-11T10:54:24.0165030Z and
2020-05-11T10:54:24.0165421Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0165933Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0166390Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0166873Z       +-org.codehaus.jackson:jackson-core-asl:1.9.13
2020-05-11T10:54:24.0167125Z and
2020-05-11T10:54:24.0167516Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0168012Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0168640Z     +-org.apache.hadoop:hadoop-common:2.8.3
2020-05-11T10:54:24.0169087Z       +-org.apache.avro:avro:1.8.2
2020-05-11T10:54:24.0169556Z         +-org.codehaus.jackson:jackson-core-asl:1.9.13
2020-05-11T10:54:24.0170071Z and
2020-05-11T10:54:24.0170510Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0171026Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0171517Z     +-org.apache.hadoop:hadoop-hdfs:2.8.3
2020-05-11T10:54:24.0172151Z       +-org.codehaus.jackson:jackson-core-asl:1.9.13
2020-05-11T10:54:24.0172551Z and
2020-05-11T10:54:24.0172985Z +-org.apache.flink:flink-end-to-end-tests:1.11-SNAPSHOT
2020-05-11T10:54:24.0173498Z   +-org.apache.flink:flink-yarn_2.11:1.11-SNAPSHOT
2020-05-11T10:54:24.0173982Z     +-org.apache.hadoop:hadoop-yarn-common:2.8.3
2020-05-11T10:54:24.0174459Z       +-org.codehaus.jackson:jackson-core-asl:1.9.13
2020-05-11T10:54:24.0174662Z 
```

Reply:
Convergence isn't just a problem for users though; we also want convergence so the tests are stable, so i don't want to throw it out in general just because Hadoop is causing yet another issue.
I'd also like to understand why the exclusions aren't working; we used them in other areas to solve convergence issues IIRC.

Reply:
But I don't know if it makes sense to mess with Hadoop's transitive dependency tree. 
In our tests, we want to ensure that Flink works with certain vanilla Hadoop versions.

If we start hand-crafting Hadoop's dependencies towards convergence, we won't ensure that Flink works with those versions -- we ensure it works with our version.

If the maven-enforcer-plugin would allow us to control the convergence check more fine-grained, I would not be opposed to it, as we need to ensure that some dependencyManagement, exclusion etc. from us is affecting vanilla Hadoop's dependency tree.

A second problem is that the exclusions might differ between the Hadoop versions we use for CI. For Hadoop 2.4.1 we have convergence, for 2.8.3 we don't.

Given these thoughts, I believe we should disable the convergence check for the tests, and rely on test failures for detecting severe issues with our Hadoop integration. We just need to accept that the Hadoop project is wild-west when it comes to dependencies.

Reply:
I'm not suggesting to converge hadoop dependencies, that is utterly futile.

What I'm suggesting is to ensure that the enforcer ignores flink-yarn but continues to run for other sub-modules, which it should because of the exclusions. Hence, it is important to understand why the exclusions aren't working here when they are working in other places.

Reply:
Okay, now I understand. I will look into it tomorrow.